### PR TITLE
build: migrate from CRA to Vite 🎉

### PR DESCRIPTION
**Pull request by wesbragagt**
Hey lovely folks 👋 , I'm a long time fan of Excalidraw and I'm working on a live workshop showing folks how to migrate a fairly large application from CRA to Vite. I chose to try it out with this project since its built with CRA.

I was very excited to see it working with Vite and decided to put together this PR for everyone to check it out and see if its something worth considering.

## Progress so far:
- [x] Development build
- [x] Production build
- [x] Environment variables
- [x] PWA support

Tests are still using react-scripts but I intend to explore getting Vitest to work.

## Test:
- `yarn start` Watch the blazingly fast HMR
- `yarn build` Pretty darn fast too
- `yarn build:preview` Build and serve a production version locally

### CRA Build before
image

### Vite Build after
image

### CODE REVIEW

**Comment by ad1992** [comment]
Wow this is great @wesbragagt 🚀
We are considering trying out Vite very soon so thanks for picking this up ❤️

**Comment by ad1992** [comment]
@wesbragagt checking the PR, the env variables don't seem to work well eg `REACT_APP_DEBUG_ENABLE_TEXT_CONTAINER_BOUNDING_BOX` is not set to true even when its set to true in `env.local`, any idea why would that happen ?

**Review by ad1992 — COMMENTED** [review]
I have reverted the ts config changes, probably those were pushed mistakenly

The flags don't work well so need to debug those

**Review by ad1992 — COMMENTED** [review]
Revert TS config changes, probably those were pushed mistakenly @wesbragagt  ?
The flags are not working well so checking that

File: src/index.tsx [hunk 1]
```diff
@@ -2,7 +2,6 @@ import { StrictMode } from "react";
 import { createRoot } from "react-dom/client";
 import ExcalidrawApp from "./excalidraw-app";
 
-import "./excalidraw-app/pwa";
```
[the remark this answers is not in the document]
Comment (wesbragagt) [reply src/index.tsx:5]: That would be setup by vite-pwa plugin automatically

**Comment by wesbragagt** [comment]
> @wesbragagt checking the PR, the env variables don't seem to work well eg `REACT_APP_DEBUG_ENABLE_TEXT_CONTAINER_BOUNDING_BOX` is not set to true even when its set to true in `env.local`, any idea why would that happen ?

Hum, it should pick that up. I will take a look. Likely the the way vite is picking up .env files.

**Comment by ad1992** [comment]
> > @wesbragagt checking the PR, the env variables don't seem to work well eg `REACT_APP_DEBUG_ENABLE_TEXT_CONTAINER_BOUNDING_BOX` is not set to true even when its set to true in `env.local`, any idea why would that happen ?
> 
> Hum, it should pick that up. I will take a look. Likely the the way vite is picking up .env files.

@wesbragagt any update on this ?

**Comment by ad1992** [comment]
> > > @wesbragagt checking the PR, the env variables don't seem to work well eg `REACT_APP_DEBUG_ENABLE_TEXT_CONTAINER_BOUNDING_BOX` is not set to true even when its set to true in `env.local`, any idea why would that happen ?
> > 
> > 
> > Hum, it should pick that up. I will take a look. Likely the the way vite is picking up .env files.
> 
> @wesbragagt any update on this ?

Rethinking on this - I think its better we migrate to `VITE_` prefix as suggested in [official docs](https://vitejs.dev/guide/env-and-mode.html) as thats recommended. The plugin [vite-plugin-env-compatible](https://github.com/IndexXuan/vite-plugin-env-compatible) I am not sure if its maintained well as its not working well with few env variables so its better we do it the `VITE` way

**Comment by ad1992** [comment]
@excalibot trigger release

**Comment by ad1992** [comment]
@excalibot trigger release

**Comment by ad1992** [comment]
As I am trying out fully to migrate to Vite, its a big change and significantly different from this PR hence I will open a separate PR so its easier to visualise the diff and keep this open as well so if the full migration doesn't work out we can still merge with current state. Hence I will also revert the vite and webpack related changes in this PR and push it in a separate PR instead

**Comment by wesbragagt** [comment]
@ad1992 Sounds like a good plan, I'm excited for the release!

**Comment by ad1992** [comment]
Vite is merged so closing this in favour of https://github.com/excalidraw/excalidraw/pull/6713

**Pull request closed without merging**